pandas/tests/dtypes/test_inference.py

```diff
@@ -1396,6 +1396,15 @@ def test_infer_dtype_period_with_na(self, na_value):
         arr = np.array([na_value, Period("2011-01", freq="D"), na_value])
         assert lib.infer_dtype(arr, skipna=True) == "period"
 
+    @pytest.mark.parametrize("na_value", [pd.NA, np.nan])
+    def test_infer_dtype_numeric_with_na(self, na_value):
+        # GH61621
+        arr = Series([1, 2, na_value], dtype=object)
+        assert lib.infer_dtype(arr, skipna=True) == "integer"
+
+        arr = Series([1.0, 2.0, na_value], dtype=object)
+        assert lib.infer_dtype(arr, skipna=True) == "floating"
```

Comment:
Nitpick:
```suggestion
        arr = np.array([1, 2, na_value], dtype=object)
        assert lib.infer_dtype(arr, skipna=True) == "integer"

        arr = np.array([1.0, 2.0, na_value], dtype=object)
        assert lib.infer_dtype(arr, skipna=True) == "floating"
```
Or
```
        ser = Series([1.0, 2.0, na_value], dtype=object)
        assert lib.infer_dtype(ser, skipna=True) == "floating"
```

Reply:
Agreed - would recommend changing the variable here to `ser`.

Reply:
Thank you both for your review. I've been missing it. I updated it. (799d9ca)